### PR TITLE
Add support for storing images

### LINKED ISSUE #63: Add images to user profiles

### PR DESCRIPTION
**Pull request by bfbachmann**
:tickets: **Ticket(s)**: Closes #63 

---

## :construction_worker: Changes

- Added support for PUT, GET, DELETE images. Current we're only able to do this for user images. All images are stored in the folder pointed to by your `IMAGE_DIR` environment variable (see README).

## :traffic_light: Concerns

Currently we don't point to these images with any extra columns in the `users` table. Perhaps we should. Not doing so makes the code simpler and it prevents situations where the information in the DB conflicts with the actual files in the `IMAGE_DIR`, so I think this is okay for now.

## :mag: Testing Instructions

`make test`. You can also use Postman to upload, retrieve, and delete images.

### CODE REVIEW

File: bounce/db/image.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+         must_exist=False):
+    """Saves the given image and returns the path to the image."""
+    path = os.path.join(directory, entity_type.value, str(entity_id),
+                        image_name)
+    if must_exist and not os.path.exists(path):
+        raise FileNotFoundError(f'No such image {path}')
+    else:
+        # Create the directory(s) to store the file in before creating it
+        os.makedirs(os.path.dirname(path), exist_ok=True)
+    with open(path, 'wb') as image_file:
```
Comment (stanleyye) [inline bounce/db/image.py]: this is synchronous right?

File: bounce/db/image.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    """Deletes the image by the given name."""
+    path = os.path.join(directory, entity_type.value, str(entity_id),
+                        image_name)
+    if os.path.exists(path):
+        os.remove(path)
+    elif must_exist:
+        raise FileNotFoundError(f'No such image {path}')
+
+
+def delete_dir(directory, entity_type, entity_id, must_exist=False):
```
Comment (stanleyye) [inline bounce/db/image.py]: to delete a directory, you must first delete all the contents inside it. Is it possible to just use the delete method here (and hence reduce some redundant code)

File: bounce/db/image.py (hunk 1 above)
Comment (bfbachmann) [reply bounce/db/image.py]: You don't store image in a database. It's bad practice.

File: bounce/db/image.py (hunk 2 above)
Comment (bfbachmann) [reply bounce/db/image.py]: Yea you're right, this is a bug. It should be `shutil.rmtree(path)`.

**Comment by bfbachmann** [comment]
@stanleyye Good point, I totally forgot to check the image type. I just added image size (1MB max) and image type (JPEG/PNG) restrictions.

**Pull request merged**